src/registry/currency/edit-validation.ts

```diff
@@ -0,0 +1,36 @@
+import {IntlShape} from 'react-intl';
+import {z} from 'zod';
+
+import {buildCommonSchema} from '@/registry/validation';
+
+// undefined (optional) for unspecified, otherwise a finite numeric value. Note that
+// null would be nicer, but formio's schema does not support null for validate.min,
+// validate.max or defaultValue
+const currencySchema = z.number().finite().optional();
+
+// case for when component.multiple=false
+const singleValueSchema = z
+  .object({multiple: z.literal(false)})
+  .and(z.object({defaultValue: currencySchema}));
+
+// case for when component.multiple=true
+const multipleValueSchema = z
+  .object({multiple: z.literal(true)})
+  .and(z.object({defaultValue: currencySchema.array()}));
+
+const defaultValueSchema = singleValueSchema.or(multipleValueSchema);
+
+const currencySpecific = z.object({
+  decimalLimit: z.number().int().positive().optional(),
```

Comment:
the types restrict this to the values 0-9 (maybe 10), so perhaps we should do the same here?

`z.union([z.literal(0), z.literal(1), z.literal(2)])` etc. (you can write a nice `Array(9).map` for this probably).

Reply:
Had to cast `as any` because it doesn't typecheck somehow